lib/GADS.pm

```diff
@@ -3024,6 +3025,99 @@ prefix '/:layout_name' => sub {
 
     };
 
+    any ['get', 'post'] => '/historic_purge/' => require_login sub {
+
+        my $layout = var('layout') or pass;
+
+        my $user        = logged_in_user;
+        my $view        = current_view($user, $layout);
+
+        my $records = GADS::Records->new(
+            user                => $user,
+            layout              => $layout,
+            schema              => schema,
+            include_children    => 1,
+            view_limit_extra_id => undef, # Override any value that may be set
+            view                => $view,
+        );
+
+        my $params = +{};
+        my $record_data = [];
+
+        if(param 'submit') {
+            my @columns = body_parameters->get_all('record');
+            $records->columns([@columns]);
+        } elsif(param 'purge') {
+            my @columns = split(/,/,body_parameters->get('records'));
+            $records->columns([@columns]);
+        }
+
+        my %ignores = (
+            id => 1,
+            createddate => 1,
+            createdby => 1,
+            serial => 1,
+        );
+
+        unless(param 'purge') {
+            while (my $record = $records->single) {
+                foreach my $column (@{$record->presentation->{columns}}) {
+                    next if $ignores{$column->{type}};
+                    my $result = +{};
+                    $result->{col_id} = $column->{id};
+                    $result->{name} = $column->{name};
+                    $result->{id} = $column->{data}->{record_id};
+                    $result->{value} = $column->{data}->{value};
+                    push @$record_data, $result;
```

Comment:
Styling: I think this would be better written as:
```
push @record_data, {
    col_id => $column->{id},
    ....
};
```
It would also be tidier to use an array and then only turn it into a reference when needed.